client/wfnews-war/src/main/angular/src/app/components/report-of-fire/compass-page/rof-compass-page.component.ts

```diff
@@ -93,21 +95,50 @@ handler(e) {
       cardinalDirection = "NW"
     }
 
-    document.getElementById("compass-face-image").style.transform = `rotate(${-compassHeading}deg)`;
-    document.getElementById("compass-heading").innerText = compassHeading.toString() + "° " + cardinalDirection;
+    if (document.getElementById("compass-face-image")) document.getElementById("compass-face-image").style.transform = `rotate(${-compassHeading}deg)`;
+    if (document.getElementById("compass-heading")) document.getElementById("compass-heading").innerText = compassHeading.toString() + "° " + cardinalDirection;
 
     this.reportOfFire.compassHeading = compassHeading;
 
+    this.useMyCurrentLocation();
+
+  } catch(err) {
+    console.error('Could not set compass heading', err)
+  }
+    
 }
 
 async useMyCurrentLocation() {
 
-  const location = await this.commonUtilityService.getCurrentLocationPromise()
-  if (location) {
-    this.currentLat = Number(location.coords.latitude);
-    this.currentLong = Number(location.coords.longitude);
+  try {
+    const location = await this.commonUtilityService.getCurrentLocationPromise()
+    if (location) {   
+      this.currentLat = this.formatDDM(Number(location.coords.latitude));
+      this.currentLong = this.formatDDM(Number(location.coords.longitude));
+    }
+
+    if (document.getElementById("location")) document.getElementById("location").innerText = this.currentLat + "," + this.currentLong;
+  } catch(err){
+    console.error('Could not find current location', err)
   }
   
 }
 
+confirmHeading() {
+  try{
+    this.reportOfFire.compassHeading = this.compassHeading;
+    this.next();
+  } catch(err){
+    console.error('Could not confirm heading', err)
+  }
+
+}
+
+formatDDM(decimal: number){
```

Comment:
A bit late, but worth noting here:

https://github.com/dhlevi/reproj-helper/blob/2cf3c3737a00a6034b8230aeb300a4cca27d6803/src/spatial-utils.ts#L63

See this for an example implementation that handles a number of corner cases